modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/Common.cs

```diff
@@ -358,5 +358,31 @@ ISymbol exportedMemberSymbol
                 typeArgumentSyntax.GetLocation(),
                 typeArgumentSyntax.SyntaxTree.FilePath));
         }
+
+        static SymbolDisplayFormat _symbolDisplayFormat = new SymbolDisplayFormat(typeQualificationStyle: SymbolDisplayTypeQualificationStyle.NameAndContainingTypesAndNamespaces);
+        public static string SyntaxToQualifiedString(SyntaxNode node, SemanticModel model, bool isFirstNode = true)
+        {
+            if (node is IdentifierNameSyntax ns && isFirstNode)
+            {
+                SymbolInfo nameInfo = model.GetSymbolInfo(ns);
+                if (nameInfo.Symbol is null) return ns.ToString();
+                return nameInfo.Symbol.ToDisplayString(_symbolDisplayFormat);
+            }
+
+            string s = "";
+            bool innerIsFirstNode = true;
+            foreach (var child in node.ChildNodesAndTokens())
+            {
+                if (child.AsNode() is SyntaxNode childNode)
+                {
+                    s += SyntaxToQualifiedString(childNode, model, isFirstNode: innerIsFirstNode);
+                    innerIsFirstNode = false;
+                }
+                else { s += child; }
+
+                s += " ";
```

Comment:
Why is adding a space needed? This seems to add unnecessary spaces to the generated syntax.

Reply:
Because otherwise we'd end up with no spaces at all in the output (which would then obviously fail to compile). I'd love to avoid the unnecessary spaces, but I haven't been able to figure out a good rule for when to add them and when not.

Reply:
You should be able to just use the existing whitespace from the nodes and tokens, it should be in the leading and trailing trivia.

Reply:
Ah, that does work, thanks!